### DISCUSSION TITLE
Error when creating Views js component

### DISCUSSION THREAD

**Discussion by pk-pankaj** [breezefront/community#90] [1. Help] [answered]
Hi @vovayatsyuk 

I am trying to create a Views component but getting errors

`component.js:579 Uncaught TypeError: Cannot read properties of undefined (reading 'length')
    at Class._applyBindings`

I have create view file with below code:

`(function () {

    $.view('myTestView', {
        component: 'Magento_Theme/js/test-view',

        create: function () {
           alert("Hi View");
        }

    });
})();`

Added in like below: 

<script type="text/x-magento-init">
    {
        "*": {
            "Magento_Theme/js/test-view": {}
        }

    }
</script>

Please help me with this.

Thanks,
Pankaj

**Comment by pk-pankaj** [comment]
The file is calling and the alert is working but still giving error which mention above

**Reply by vovayatsyuk** [reply, marked as answer]
The issue is caused by your View not being attached to any DOM element. Usually, there should be a DOM element with a `scope` attribute to make it work (`data-bind="scope: 'wishlist'"`)

Since you do not render anything/using DOM you don't need a view or widget, I think you should use raw js file in this case. Take a look at [theme.js for example](https://github.com/breezefront/theme-frontend-breeze-blank/blob/master/web/js/breeze/theme-blank.js#L53).

**Reply by pk-pankaj** [reply]
Yes issue was due to scope.

Thanks
